src/skmultiflow/data/temporal_data_stream.py

```diff
@@ -0,0 +1,148 @@
+            # order sample_delay, check if not single delay
+            if not self.single_delay:
+                self.sample_delay[np.argsort(self.time)]
+            # order self.time
+            self.time = self.time.sort_values()
+        super().__init__(data, y, target_idx, n_targets, cat_features, name)
+    
+    # get next sample, returning sample_x, sample_time and sample_y
+    def next_sample(self, batch_size=1):
+        """ next_sample
+        If there is enough instances to supply at least batch_size samples, those
+        are returned. If there aren't a tuple of (None, None) is returned.
+        Parameters
+        ----------
+        batch_size: int
+            The number of instances to return.
+        Returns
+        -------
+        tuple or tuple list
+            Returns the next batch_size instances (sample_x, sample_time, sample_y, sample_weight (if available), sample_delay (if available)).
+            For general purposes the return can be treated as a numpy.ndarray.
+        """
+        self.sample_idx += batch_size
+        
+        try:
+
+            self.current_sample_x = self.X[self.sample_idx - batch_size:self.sample_idx, :]
+            self.current_sample_y = self.y[self.sample_idx - batch_size:self.sample_idx, :]
+            self.current_sample_time = self.time[self.sample_idx - batch_size:self.sample_idx]
+
+            # check if its a single delay
+            if self.single_delay:
+                # create list with same delay for each instance
+                self.current_sample_delay = np.full(batch_size, self.sample_delay)
+            else:
+                # get delays for each instance
+                self.current_sample_delay = self.sample_delay[self.sample_idx - batch_size:self.sample_idx]
+
+            if self.n_targets < 2:
+                self.current_sample_y = self.current_sample_y.flatten()
+
+            # create base output
+            output = [self.current_sample_x, self.current_sample_time, self.current_sample_delay, self.current_sample_y]
+
+            # check if sampe_weight is available
+            if self.sample_weight is not None:
+                self.current_sample_weight = self.sample_weight[self.sample_idx - batch_size:self.sample_idx, :]
+                # add to output
+                output.append(self.current_sample_weight)
+
+        except IndexError:
+            self.current_sample_x = None
+            self.current_sample_y = None
+            self.current_sample_time = None
+            self.current_sample_delay = None
+
+            # create base output
+            output = [self.current_sample_x, self.current_sample_time, self.current_sample_delay, self.current_sample_y]
+            # check if sampe_weight is available
+            if self.sample_weight is not None:
+                self.current_sample_weight = None
+                output.append(self.current_sample_weight)
+            
+        return output
```

Comment:
All methods in generator files such as `random_tree_generator.py` return a tuple of `current_sample_x` and `current_sample_y`. I think you will lead to an inconsistency in the API (`base_stream.py`) by returning four elements. I'd respect the same semantic and add an extra method for the **temporal data stream** .   

Also, should return a tuple instead of a list.

Reply:
But in this case, how will we get the temporal information (and also weights)? Any thoughts on how to implement this?

Reply:
@fabriciojoc
Good question. Let me check the design and the usage first!

Reply:
@fabriciojoc
Sorry for the late reply! I think we can keep it this way.

Reply:
Thanks @garawalid!

Reply:
I'd return the result (`output`) as a tuple, so we can ignore unused elements with `_`.
Example: 
```python
current_sample_x, current_sample_time, _, _, _ = tds.next_sample()
```

Reply:
Thanks @garawalid and @jacobmontiel!

Reply:
Updated in commit [aee8c61c44f147404d1ec11950508fe9374c8fb3](https://github.com/scikit-multiflow/scikit-multiflow/commit/aee8c61c44f147404d1ec11950508fe9374c8fb3)!